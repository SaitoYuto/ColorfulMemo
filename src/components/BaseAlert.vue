<template>
  <v-alert closable :color="color" :text="text" :variant="variant"></v-alert>
</template>

<script lang="ts" setup>
import { ALERT_VARIANT } from "@/constants/AlertVariant";
import type { AlertVariant } from "@/types/AlertVariant";
defineProps({
  closable: {
    type: Boolean,
    default: true,
  },
  color: {
    type: String,
    default: undefined,
  },
  text: {
    type: String,
    required: true,
  },
  variant: {
    type: String as PropType<AlertVariant>,
    default: ALERT_VARIANT.OUTLINED,
  },
});
</script>
