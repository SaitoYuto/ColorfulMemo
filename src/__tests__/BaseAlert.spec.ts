import BaseAlert from "@/components/BaseAlert.vue";
import { mount } from "@vue/test-utils";
import "@/__tests__/setup";

describe("BaseAlert.vue", () => {
  it("renders props.text when passed", async () => {
    const text = "Alert Text";
    const wrapper = mount(BaseAlert, {
      props: {
        text: text,
      },
    });
    expect(wrapper.props("text")).toBe(text);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders props.text when an empty string passed", async () => {
    const emptyText = "";
    const wrapper = mount(BaseAlert, {
      props: {
        text: emptyText,
      },
    });
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.props("text")).toBe(emptyText);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
