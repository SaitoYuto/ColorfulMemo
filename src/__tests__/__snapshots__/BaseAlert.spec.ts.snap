// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseAlert.vue > renders props.text when an empty string passed 1`] = `
"<div class="v-alert v-theme--light v-alert--density-default v-alert--variant-outlined" role="alert">
  <!----><span class="v-alert__underlay"></span>
  <!---->
  <div class="v-alert__prepend"><i class="mdi-information mdi v-icon notranslate v-theme--light" style="font-size: 28px; height: 28px; width: 28px;" aria-hidden="true" density="default"></i></div>
  <div class="v-alert__content">
    <!---->
    <!---->
  </div>
  <!---->
  <div class="v-alert__close"><button type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-x-small v-btn--variant-text" aria-label="Close"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
      <!---->
      <!---->
    </button></div>
</div>"
`;

exports[`BaseAlert.vue > renders props.text when passed 1`] = `
"<div class="v-alert v-theme--light v-alert--density-default v-alert--variant-outlined" role="alert">
  <!----><span class="v-alert__underlay"></span>
  <!---->
  <div class="v-alert__prepend"><i class="mdi-information mdi v-icon notranslate v-theme--light" style="font-size: 28px; height: 28px; width: 28px;" aria-hidden="true" density="default"></i></div>
  <div class="v-alert__content">
    <!---->Alert Text
    <!---->
  </div>
  <!---->
  <div class="v-alert__close"><button type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-x-small v-btn--variant-text" aria-label="Close"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
      <!---->
      <!---->
    </button></div>
</div>"
`;
